File: ModuleCSharp/Tool.cs
using FarNet;

namespace ModuleCSharp;

[ModuleTool(Name = "ModuleCSharp", Options = ModuleToolOptions.AllMenus, Id = "f584e496-b801-4df3-be4f-2848d3a6e77c")]
public class Tool : ModuleTool
{
	public override void Invoke(object sender, ModuleToolEventArgs e)
	{
		Far.Api.Message($"Hello from {e.From}");
	}
}

File: ModuleFSharp/Tool.fs
namespace ModuleFSharp
open FarNet

[<ModuleTool(Name = "ModuleFSharp", Options = ModuleToolOptions.AllMenus, Id = "cc0ecb2f-552a-4fe4-90f0-227077386890")>]
type Tool1() =
    inherit ModuleTool()

    override _.Invoke(_, e) =
        Far.Api.Message $"Hello from {e.From}"
